wordpress-mobile/WordPress-Android
# [Gutenberg] Add `enableCaching` param to fetch request in edit post activity

`gutenberg` PR: https://github.com/WordPress/gutenberg/pull/31186
`gutenberg-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/3419

This PR introduces the param `enableCaching` in the fetch request method of the edit post activity to allow Gutenberg to control caching of the network requests.

**To test:**
Check testing instructions in the [`gutenberg` PR](https://github.com/WordPress/gutenberg/pull/31186).

## Regression Notes
1. Potential unintended areas of impact

These changes should only affect Gutenberg-mobile so I don't foresee any potential unintended areas of impact apart from that one.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

I verified that the requests made from Gutenberg-mobile were cached or not depending on the `enableCaching` param.

Additionally, this code is tested within this [unit test](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/test/java/org/wordpress/android/ui/posts/reactnative/ReactNativeRequestHandlerTest.kt).

3. What automated tests I added (or what prevented me from doing so)

None.

PR submission checklist:
